From 5cea2caf937019f08ad45cff0da832078e34beeb Mon Sep 17 00:00:00 2001
From: "Singh, Sapna1" <sapna1.singh@intel.com>
Date: Fri, 26 Aug 2022 12:30:39 +0530
Subject: [PATCH] Video Recording resolution mismatch during multi-user switch

CameraId of the user was not getting converted
to MultiCameraId which leads to reading the
cameraId 0's supported video quality list instead
of querying for specific user's cameraId

Convert user's cameraId to MultiCameraId first and
then use this cameraId for querying the video Quality
supported list

Tracked-On: OAM-103573
Change-Id: I2cf2e50a4ee2a602cdfe6e10982392eaf16db5a2
Signed-off-by: Singh, Sapna1 <sapna1.singh@intel.com>
---
 src/com/android/camera/settings/SettingsUtil.java | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/com/android/camera/settings/SettingsUtil.java b/src/com/android/camera/settings/SettingsUtil.java
index ffbdcd5f4..374f67c7d 100644
--- a/src/com/android/camera/settings/SettingsUtil.java
+++ b/src/com/android/camera/settings/SettingsUtil.java
@@ -42,6 +42,9 @@ import java.util.Collections;
 import java.util.Comparator;
 import java.util.LinkedList;
 import java.util.List;
+import android.hardware.camera2.CameraManager;
+import com.android.camera.util.AndroidServices;
+import android.hardware.camera2.CameraAccessException;
 
 /**
  * Utility functions around camera settings.
@@ -383,6 +386,7 @@ public class SettingsUtil {
         }
 
         OneCameraManager cameraManager;
+        CameraManager mcameraManager = AndroidServices.instance().provideCameraManager();
         try {
             cameraManager = OneCameraModule.provideOneCameraManager();
         } catch (OneCameraException e) {
@@ -391,7 +395,8 @@ public class SettingsUtil {
         }
 
         try {
-            CameraId camId = CameraId.from(Integer.toString(cameraId));
+            String[] cameraIdList = mcameraManager.getCameraIdList();
+            CameraId camId = CameraId.from(cameraIdList[cameraId]);
             List<Size> supportedPreviewSizes = (cameraManager.getOneCameraCharacteristics(camId)
                                 .getSupportedPreviewSizes());
             int height = 0;
@@ -416,11 +421,13 @@ public class SettingsUtil {
                     if (previewSize.getHeight() == height) return true;
                 }
             }
+        } catch (CameraAccessException e) {
+            Log.e(TAG, "Unable to get CameraId List " + e.getMessage());
+            return false;
         } catch (OneCameraAccessException e) {
             Log.d(TAG, "video quality check failed:  " + e.getMessage());
             return false;
         }
-
         return false;
     }
 
-- 
2.37.0

